lib/osf-components/addon/components/contributors/user-search/widget/template.hbs

```diff
@@ -6,13 +6,14 @@
             @type='text'
             @placeholder={{t 'registries.registration_metadata.add_contributors.search_placeholder'}}
             @value={{this.query}}
-            @enter={{perform this.fetchUsers}}
+            @enter={{queue this.resetPage (perform this.fetchUsers)}}
+            {{on 'input' this.resetPage}}
```

Comment:
afaik, on`input` event is fired right after the value of the input element has changed, so I think `on 'input' this.resetPage` is all you need to ensure page is set to 1? if the user proceeds to hit enter or click search button afterwards `currentUsersPage` should already be set to 1.

Reply:
If you click the search button a second time without entering any more text, wouldn't it error in that case? I can see how it might not be needed for the `@enter` case, however.

Reply:
The `resetPage` invoked here is for when the user hits the search button/enter multiple times.

Reply:
hm, button mashing is handled well by `ember-concurrency`; the `fetchUsers` task should be a restartable task. See http://ember-concurrency.com/docs/task-concurrency. For input search type of tasks, we have ample examples in the codebase of using a combo of restartable task and debouncing to provide a smooth UX.

Reply:
True, but every time you search, it increments the page counter in anticipation of the next search. So just hitting the button twice would cause it to search page 2, and then again would cause it to search page 3. So one needs to reset the counter before the search is performed (unless you're explicitly getting the next page). Another potential option is to have a "load more users" task that increments the counter then fetches users, rather than just automatically incrementing the page in the `fetchUsers` task.

Reply:
Right, I think I understand better what's going on. So clicking the search button several for the same query was intended to serve as the only way the user navigates to other search results pages? @adlius @brianjgeiger  is this right?

It's not a consistent at all with other search/list views/pages, also I presume some kind of accessibility hazard.

Reply:
That's a side effect which I *believe* this.resetPage is attempting to prevent. The actual user search list has

```
        {{#if @manager.shouldShowLoadMoreUsers}}
            <div local-class='InfinityLoader'
                {{in-viewport
                    onEnter=(fn @manager.fetchUsers)
                }}
            >
                {{t 'osf-components.contributors.loadMore'}}
            </div>
        {{/if}}
```

which is the thing that should actually be fetching more pages. This looks to me like the thing that could call a method to increment the page if we wanted to not do the `this.resetPage`.

Reply:
If the user hits the enter/search button multiple times while the search term remains the same, the API will fetch for the next page every time the user hits the button. This is something we are trying to prevent by resetting page number back to 1 upon clicking search/enter.

Reply:
Okay. I got this. but @adlius  I have another confusion point in the user-search list component the passed in `@manager` has no `fetchUsers` task nor `shouldShowLoadMoreUsers`, `isLoadingUsers` bools. I must be missing something, it's Friday afternoon.

https://github.com/CenterForOpenScience/ember-osf-web/blob/release/21.1.0/lib/osf-components/addon/components/contributors/user-search/list/template.hbs#L22-L30

Reply:
Coming back to the button-mashing issue here: if I type in the user I am searching and mash the enter key afterward, it's going to create a backlog of `fetchUsers` and these task instances will be looking for the 2nd page, since `this.currentUsersPage` is incremented in `fetchUsers`. I think as @fabmiz suggested, `fetchUsers` should be `restartable/drop: true` (my preference is for drop, since restartable will prevent it from running to completion if you're mashing enter). As for getting the subsequent pages, is it possible to increment `this.currentUsersPage` before calling `fetchUsers` when the infinity loader comes into the viewport? (https://github.com/CenterForOpenScience/ember-osf-web/blob/release/21.1.0/lib/osf-components/addon/components/contributors/user-search/list/template.hbs#L22-L30)